flexmeasures/data/tests/test_train_predict_pipeline.py

```diff
@@ -203,6 +203,31 @@ def test_train_predict_pipeline(
             source
         ), "string representation of the Forecaster (DataSource) should mention the used model"
 
+        assert isinstance(pipeline_returns, list), "pipeline should return a list"
+        assert all(
+            isinstance(item, dict) for item in pipeline_returns
+        ), "each item should be a dict"
+        for index, pipeline_return in enumerate(pipeline_returns):
+            if not dg_params["as_job"]:
+                assert {"data", "sensor"}.issubset(
+                    pipeline_return.keys()
+                ), "returned dict should have data and sensor keys"
+                assert (
+                    pipeline_return["sensor"].id == dg_params["sensor_to_save"].id
+                ), "returned sensor should match sensor that forecasts will be saved into"
+                try:
+                    pd.testing.assert_frame_equal(
+                        forecasts.sort_index(),
+                        pipeline_return["data"].sort_index(),
+                    )
+                except AssertionError as e:
+                    raise AssertionError(
+                        f"returned data should match stored forecasts: {e}"
+                    )
```

Comment:
These try/except blocks remain a bit of a code smell for me. For one, it swallows the line number of the failing assert, making it more cumbersome for future devs to debug.

In some cases, though, I do get that you'd want to add some more context to a lower-level error. In those cases, make sure you propagate the original error. Here that could have been done with `raise AssertionError("extra context blabla") from e`. The `raise ... from ...` preserves the context of the original exception.

But in this particular case, I'd really just float the original error instead, so the error log would get straight to the point, by removing the try/except block surrounding the dataframe comparison.